contracts/voting_escrow_delegation/tests/integration.rs

```diff
@@ -0,0 +1,1151 @@
+                        percentage: Uint128::new(100),
+                        expire_time: WEEK,
+                        token_id: "token_1".to_string(),
+                        recipient: "user2".to_string(),
+                    },
+                    &[],
+                )
+                .unwrap();
+
+            // try to create delegation with the same token ID
+            let err = router_ref
+                .execute_contract(
+                    Addr::unchecked("user"),
+                    delegator_helper.delegation_instance.clone(),
+                    &ExecuteMsg::CreateDelegation {
+                        percentage: Uint128::new(100),
+                        expire_time: WEEK,
+                        token_id: "token_1".to_string(),
+                        recipient: "user2".to_string(),
+                    },
+                    &[],
+                )
+                .unwrap_err();
+            assert_eq!(
+                "A delegation with a token token_1 already exists.",
+                err.root_cause().to_string()
+            );
+
+            // try create delegation without free voting power
+            let err = router_ref
+                .execute_contract(
+                    Addr::unchecked("user"),
+                    delegator_helper.delegation_instance.clone(),
+                    &ExecuteMsg::CreateDelegation {
+                        percentage: Uint128::new(30),
+                        expire_time: WEEK,
+                        token_id: "token_2".to_string(),
+                        recipient: "user2".to_string(),
+                    },
+                    &[],
+                )
+                .unwrap_err();
+            assert_eq!(
+                "You have already delegated all the voting power.",
+                err.root_cause().to_string()
+            );
+
+            // check user's nft tokens
+            let resp = nft_helper
+                .tokens(&router_ref.wrap().into(), "user", None, None)
+                .unwrap();
+            assert_eq!(EMPTY_TOKENS, resp.tokens);
+
+            // check user's adjusted balance
+            let resp = router_ref
+                .wrap()
+                .query::<Uint128>(&QueryRequest::Wasm(WasmQuery::Smart {
+                    contract_addr: delegator_helper.delegation_instance.to_string(),
+                    msg: to_binary(&QueryMsg::AdjustedBalance {
+                        account: "user".to_string(),
+                        timestamp: None,
+                    })
+                    .unwrap(),
+                }))
+                .unwrap();
+            assert_eq!(Uint128::new(0), resp);
+
+            // check user2's received nft tokens
+            let resp = nft_helper
+                .tokens(&router_ref.wrap().into(), "user2", None, None)
+                .unwrap();
+            assert_eq!(vec!["token_1"], resp.tokens);
+
+            // check user2's adjusted balance
+            let resp = router_ref
```

Comment:
Check user's VP before and after delegation created. Calculate, that a user has lost 30% of his VP. Then after delegation check that adjusted balance + total delegate VP = VP before delegation.